website/content/docs/agent/template.mdx

```diff
@@ -226,16 +238,32 @@ this by inspecting the secret's time-to-live (TTL).
 
 ### Certificates
 
-If a secret is a [certificate](/docs/secrets/pki), Vault Agent template will fetch the new certificate
-using the certificates `validTo` field.
+#### Rendering using the `secret` template function
+
+If a [certificate](/docs/secrets/pki) is rendered using the `secret` template
+function, Vault Agent template will have the following fetching and re-rendering
+behaviors on certificates:
+
+- Fetches a new certificate on Agent startup, even if previously rendered
+certificates are still valid.
+- If `generate_lease` is unset or set to `false`, it uses the certificate's
+`validTo` field to determine re-fetch interval.
+- If `generate_lease` is set to `true`, apply the non-renewable, leased secret
+rules.
+- On Agent's auto-auth re-authentication, due to a token expiry for example, it
+fetches and re-renders a new certificate even if the existing certificate is
+valid.
+
+#### Rendering using the `pkiCert` template function
```

Comment:
Am I right thinking this is just better, and we should recommend this way? If so, maybe worth putting it first, and including a sentence to say it's the recommended way to fetch certificates with Agent

Reply:
Yeah, this would be better for certificates for most cases. I'll get this switched and add a note.